src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -692,9 +710,25 @@ else if (message.isDDLMessage()) {
                                     offsetContext.updateRecordPosition(part, lsn, lastCompletelyProcessedLsn, message.getRawCommitTime(),
                                             String.valueOf(message.getTransactionId()), tableId, message.getRecordTime());
 
-                                    boolean dispatched = message.getOperation() != Operation.NOOP
-                                            && dispatcher.dispatchDataChangeEvent(part, tableId, new YugabyteDBChangeRecordEmitter(part, offsetContext, clock, connectorConfig,
-                                                    schema, connection, tableId, message, pgSchemaNameInRecord, tabletId, taskContext.isBeforeImageEnabled()));
+
+                                    boolean dispatched = message.getOperation() != Operation.NOOP;
+                                    if (connectorConfig.isYSQLDbType()) {
+                                        dispatched = dispatched
+                                                && dispatcher.dispatchDataChangeEvent(part, tableId,
+                                                        new YugabyteDBChangeRecordEmitter(part, offsetContext, clock,
+                                                                connectorConfig,
+                                                                schema, connection, tableId, message,
+                                                                pgSchemaNameInRecord, tabletId,
+                                                                taskContext.isBeforeImageEnabled()));
+                                    } else {
+                                        dispatched = dispatched
+                                                && dispatcher.dispatchDataChangeEvent(part, tableId,
```

Comment:
We seem to be doing the same thing in both the blocks, perhaps use a conditional operator to pass arguments?

```
 dispatched = dispatched
                                                && dispatcher.dispatchDataChangeEvent(part, tableId,
                                                        new YugabyteDBChangeRecordEmitter(part, offsetContext, clock,
                                                                connectorConfig,
                                                                schema, connection, tableId, message,
                                                                connectorConfig.isYSQLDbType() ? pgSchemaNameInRecord : tableId.catalog(),
                                                                tabletId,
                                                                taskContext.isBeforeImageEnabled()));
```